diff --git a/dwm.c b/dwm.c
index 4465af1..fe22a9e 100644
--- a/dwm.c
+++ b/dwm.c
@@ -966,7 +966,18 @@ grabkeys(void)
 void
 incnmaster(const Arg *arg)
 {
-	selmon->nmaster = MAX(selmon->nmaster + arg->i, 0);
+	int num_clients = 0;
+	Client* cl = selmon->clients;
+	int nmaster = MAX(selmon->nmaster + arg->i, 0);
+
+	while(cl) {
+		num_clients++;
+		cl = cl->next;
+	}
+
+	if(nmaster > num_clients) return;
+
+	selmon->nmaster = nmaster;
 	arrange(selmon);
 }
 
